### DISCUSSION TITLE
installing Vite Ruby in a Rails 7 app with importmaps

### DISCUSSION THREAD

**Discussion by quaasi** [ElMassimo/vite_ruby#233] [Q&A] [answered]
I have a Rails 7 app using importmaps. So no JS bundling tool needed so far. Will I have to switch from importmaps to say es-build in order to get Vite Ruby running? 
I followed the [setup guide](https://vite-ruby.netlify.app/guide/) and see the "Vite ⚡️ Rails" in the browser console. But no JS is loaded anymore.

**Comment by ElMassimo** [comment, marked as answer]
Hi Michael!

If you are referring to [`importmap-rails`](https://github.com/rails/importmap-rails), you typically wouldn't combine it with Vite Ruby.

- Vite is meant to [bundle in production](https://vitejs.dev/guide/why.html#why-bundle-for-production) for better performance
-  [`importmap-rails`](https://github.com/rails/importmap-rails) is only useful if you don't want to use a tool like Webpack or Vite
- Installing npm packages is a nicer experience than pinning packages manually in [`importmap-rails`](https://github.com/rails/importmap-rails)

If for some reason you don't want to bundle certain libraries, you could try an [import map plugin for Vite](https://github.com/pakholeung37/vite-plugin-import-maps), but it's unusual, and I recommend that you try using plain Vite first.
